--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/Makefile.am	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/Makefile.am	2008-07-05 23:45:34.250000000 -0500
@@ -1,19 +1,17 @@
 amarokrubylibdir = $(libdir)/ruby_lib
 
-amarokrubylib_LTLIBRARIES = libhttp11.la
+amarokrubylib_LTLIBRARIES = http11.la
 
 amarokrubylib_DATA = http11.rb
 
 INCLUDES = -I$(ruby_includes) -Dinline=__inline__ -Dasm=__asm__
 
-libhttp11_la_LDFLAGS = \
-	-shared \
-    $(ruby_ldflags)
+http11_la_LDFLAGS = -module $(KDE_PLUGIN) $(ruby_ldflags) -shrext .so
 
 #http11_so_LIBADD = \
 #    -lruby18
 
-libhttp11_la_SOURCES = \
+http11_la_SOURCES = \
     http11.c \
     http11_parser.c \
     tst_cleanup.c \
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.c	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.c	2008-07-05 23:45:34.265625000 -0500
@@ -526,7 +526,7 @@
 }
 
 
-void Init_libhttp11()
+void Init_http11()
 {
 
   mMongrel = rb_define_module("Mongrel");
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.rb	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.rb	2008-07-05 23:45:34.265625000 -0500
@@ -1,2 +1,2 @@
-require 'libhttp11'
+require 'http11.so'
 
