--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/mtp/mtpmediadevice.cpp	2008-07-05 23:45:10.281250000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/mtp/mtpmediadevice.cpp	2008-07-05 23:45:34.281250000 -0500
@@ -1287,7 +1287,7 @@
 MtpMediaDevice::rmbPressed( QListViewItem *qitem, const QPoint &point, int )
 {
 
-    enum Actions {RENAME, DOWNLOAD, DELETE, MAKE_PLAYLIST, UPDATE_ALBUM_ART};
+    enum Actions {RENAME, DOWNLOAD, _DELETE, MAKE_PLAYLIST, UPDATE_ALBUM_ART};
 
     MtpMediaItem *item = static_cast<MtpMediaItem *>( qitem );
     if( item )
@@ -1309,7 +1309,7 @@
             break;
         }
 
-        menu.insertItem( SmallIconSet( Amarok::icon( "remove" ) ), i18n( "Delete from device" ), DELETE );
+        menu.insertItem( SmallIconSet( Amarok::icon( "remove" ) ), i18n( "Delete from device" ), _DELETE );
 
         int id =  menu.exec( point );
         switch( id )
@@ -1322,7 +1322,7 @@
                 newPlaylist( name, m_playlistItem, items );
             }
             break;
-        case DELETE:
+        case _DELETE:
             MediaDevice::deleteFromDevice();
             break;
         case RENAME:
