--- origsrc/amarok-1.86/src/metadata/tfile_helper.cpp	2008-07-21 12:06:42.000000000 -0500
+++ src/amarok-1.86/src/metadata/tfile_helper.cpp	2008-07-29 01:15:08.968750000 -0500
@@ -26,7 +26,7 @@
 #include <strings.h>
 #include <wchar.h>
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__CYGWIN__)
 int wcscasecmp(const wchar_t *s1, const wchar_t *s2)
 {
      int i;
--- origsrc/amarok-1.86/src/strigi/audible/audiblethroughanalyzer.cpp	2008-07-21 12:06:22.000000000 -0500
+++ src/amarok-1.86/src/strigi/audible/audiblethroughanalyzer.cpp	2008-07-29 10:02:25.531250000 -0500
@@ -197,7 +197,7 @@
 
     analysisResult->addValue(mimeTypeField, "audio/audible");
     analysisResult->addValue(audioDurationField, readBigEndianUInt32(buf+61));
-    analysisResult->addValue(channelsField, 1);
+    analysisResult->addValue(channelsField, uint32_t(1));
 
     return in;
 }
--- origsrc/amarok-1.86/src/strigi/mp4/mp4throughanalyzer.cpp	2008-07-21 12:06:22.000000000 -0500
+++ src/amarok-1.86/src/strigi/mp4/mp4throughanalyzer.cpp	2008-07-29 10:52:45.156250000 -0500
@@ -317,15 +317,15 @@
    std::string format(buf+12,4);
    if(format == "mp4v" || format == "avc1" || format == "encv" || format == "s263")
    {
-      analysisResult->addValue(imageWidthField, readBigEndianUInt16(buf+40));
-      analysisResult->addValue(imageHeightField, readBigEndianUInt16(buf+42));
+      analysisResult->addValue(imageWidthField, readBigEndianUInt32(buf+40));
+      analysisResult->addValue(imageHeightField, readBigEndianUInt32(buf+42));
       analysisResult->addValue(videoCodecField, format);
    }
    else  if(format == "mp4a" || format == "enca" || format == "samr" || format == "sawb") 
    {
-      uint16_t channels = readBigEndianUInt16(buf+32);
+      uint32_t channels = readBigEndianUInt32(buf+32);
       analysisResult->addValue(channelsField, channels);
-      uint64_t samplesize = readBigEndianUInt16(buf+34);
+      uint64_t samplesize = readBigEndianUInt32(buf+34);
       std::stringstream stream;
       stream << samplesize << " bit int";
       analysisResult->addValue(sampleFormatField, stream.str());
@@ -494,7 +494,7 @@
       }
       else if(type == "rtng")
       {
-         analysisResult->addValue(userRatingField, readBigEndianUInt16(buf+8));
+         analysisResult->addValue(userRatingField, readBigEndianUInt32(buf+8));
       }
       else if(type == "cpil")
       {
