--- origsrc/amarok-1.4.9.1/amarok/src/Makefile.am	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/Makefile.am	2008-07-05 23:45:34.234375000 -0500
@@ -165,6 +165,7 @@
 
 libamarok_la_LDFLAGS = \
     $(all_libraries) \
+	-no-undefined \
     $(KDE_RPATH)
 
 METASOURCES = \
@@ -189,12 +190,7 @@
 amarokapp_SOURCES = main.cpp
 
 amarokapp_LDADD = \
-    $(top_builddir)/amarok/src/amarokcore/libamarokcore.la \
     libamarok.la \
-    $(top_builddir)/amarok/src/analyzers/libanalyzers.la \
-    $(top_builddir)/amarok/src/plugin/libplugin.la \
-    $(top_builddir)/amarok/src/statusbar/libstatusbar.la \
-    $(top_builddir)/amarok/src/metadata/libmetadata.la \
     $(LIB_KUTILS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KHTML) $(LIB_KNEWSTUFF) \
     $(TAGLIB_LIBS) $(gl_libs) $(LIB_SQLITE) $(LIB_SQLITE_LOCAL) $(EXSCALIBAR_LIBS) \
     $(LIB_TUNEPIMP) \
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/Makefile.am	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/Makefile.am	2008-07-05 23:45:34.250000000 -0500
@@ -1,19 +1,17 @@
 amarokrubylibdir = $(libdir)/ruby_lib
 
-amarokrubylib_LTLIBRARIES = libhttp11.la
+amarokrubylib_LTLIBRARIES = http11.la
 
 amarokrubylib_DATA = http11.rb
 
 INCLUDES = -I$(ruby_includes) -Dinline=__inline__ -Dasm=__asm__
 
-libhttp11_la_LDFLAGS = \
-	-shared \
-    $(ruby_ldflags)
+http11_la_LDFLAGS = -module $(KDE_PLUGIN) $(ruby_ldflags) -shrext .so
 
 #http11_so_LIBADD = \
 #    -lruby18
 
-libhttp11_la_SOURCES = \
+http11_la_SOURCES = \
     http11.c \
     http11_parser.c \
     tst_cleanup.c \
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.c	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.c	2008-07-05 23:45:34.265625000 -0500
@@ -526,7 +526,7 @@
 }
 
 
-void Init_libhttp11()
+void Init_http11()
 {
 
   mMongrel = rb_define_module("Mongrel");
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.rb	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/daap/mongrel/http11/http11.rb	2008-07-05 23:45:34.265625000 -0500
@@ -1,2 +1,2 @@
-require 'libhttp11'
+require 'http11.so'
 
--- origsrc/amarok-1.4.9.1/amarok/src/mediadevice/mtp/mtpmediadevice.cpp	2008-07-05 23:45:10.281250000 -0500
+++ src/amarok-1.4.9.1/amarok/src/mediadevice/mtp/mtpmediadevice.cpp	2008-07-05 23:45:34.281250000 -0500
@@ -1287,7 +1287,7 @@
 MtpMediaDevice::rmbPressed( QListViewItem *qitem, const QPoint &point, int )
 {
 
-    enum Actions {RENAME, DOWNLOAD, DELETE, MAKE_PLAYLIST, UPDATE_ALBUM_ART};
+    enum Actions {RENAME, DOWNLOAD, _DELETE, MAKE_PLAYLIST, UPDATE_ALBUM_ART};
 
     MtpMediaItem *item = static_cast<MtpMediaItem *>( qitem );
     if( item )
@@ -1309,7 +1309,7 @@
             break;
         }
 
-        menu.insertItem( SmallIconSet( Amarok::icon( "remove" ) ), i18n( "Delete from device" ), DELETE );
+        menu.insertItem( SmallIconSet( Amarok::icon( "remove" ) ), i18n( "Delete from device" ), _DELETE );
 
         int id =  menu.exec( point );
         switch( id )
@@ -1322,7 +1322,7 @@
                 newPlaylist( name, m_playlistItem, items );
             }
             break;
-        case DELETE:
+        case _DELETE:
             MediaDevice::deleteFromDevice();
             break;
         case RENAME:
--- origsrc/amarok-1.4.9.1/amarok/src/metadata/mp4/mp4tag.cpp	2008-04-10 15:16:37.000000000 -0500
+++ src/amarok-1.4.9.1/amarok/src/metadata/mp4/mp4tag.cpp	2008-07-05 23:45:34.296875000 -0500
@@ -77,7 +77,7 @@
     uint8_t boolvalue;
     uint16_t numvalue, numvalue2;
     uint8_t *image;
-    uint32_t imageSize;
+    u_int32_t imageSize;
     if (MP4GetMetadataName(mp4file, &value) && value != NULL) {
         m_title = String(value, String::UTF8);
         free(value);
